Frameworks/UIKit/UIControl.mm

```diff
@@ -221,7 +221,7 @@ - (NSSet*)allTargets {
 */
 - (void)setEnabled:(BOOL)enabled {
     // only update the value and then relayout if value actually changed
-    if ((_curState & UIControlStateDisabled) != enabled) {
+    if ((_curState & UIControlStateDisabled) != !enabled) {
```

Comment:
This doesn't seem like a type-safe comparison. `_curState & x` evaluates to an int, not a boolean. There's probably a cleaner way to structure this check.